tests/test_publisher.py

```diff
@@ -0,0 +1,98 @@
+import os
+import queue
+import threading
+from itertools import chain
+
+import pytest
+
+from lektor.publisher import _join_pipes_using_select
+from lektor.publisher import _join_pipes_using_threads
+from lektor.publisher import Command
+
+
+def test_Command_triggers_no_warnings(recwarn):
+    # This excercises the issue where publishing via rsync resulted
+    # in ResourceWarnings about unclosed streams.
+
+    # This is essentially how RsyncPublisher runs rsync.
+    with Command(["echo"], env={}) as client:
+        for _ in client:
+            pass
+    del client
```

Comment:
Is the `del` call necessary here? I don't think Python gives any guarantees about when the destructor runs.

Reply:
The unclosed stream ResourceWarnings don't get emitted until the streams are garbage collected.  That happens when the Popen is collected, which, in turn, happens with the Command is collected.
Without the `del`, that doesn't happen until the test is completed, in which case, the ResourceWarnings (should there be any) don't get caught in the recwarn.